src/Microsoft.Extensions.Primitives/StringSegment.cs

```diff
@@ -428,26 +364,28 @@ public StringSegment Subsegment(int offset, int length)
         /// <param name="start">The zero-based index position at which the search starts. </param>
         /// <param name="count">The number of characters to examine.</param>
         /// <returns>The zero-based index position of <paramref name="c"/> from the beginning of the <see cref="StringSegment"/> if that character is found, or -1 if it is not.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public int IndexOf(char c, int start, int count)
         {
-            if (start < 0 || Offset + start > Buffer.Length)
+            var offset = Offset + start;
+
+            if (!HasValue || start < 0 || offset > Buffer.Length)
```

Comment:
`Offset + start` could have overflowed and become negative. Change to unsigned compare?

`|| (uint)offset > (uint)Buffer.Length`

Reply:
Good point. Will be added when I update the PR. A test, that covers this, is also added.

Although this overflow-case was handled by `Buffer.IndexOf(c, offset, count)` (i.e. string.IndexOf) it's better to handle this explicit and have the correct arg-name show in the exception-message.